lib/watir/elements/element.rb

```diff
@@ -495,6 +479,48 @@ def browser
 
     protected
 
+    def wait_for_exists(timeout = nil)
+      return assert_exists unless Watir.relaxed_locate?
+
+      timeout ||= Watir.default_timeout
+      return if exists? # Performance shortcut
+
+      assert_element_found unless Watir.always_locate? # don't re-lookup element
+      timeout_off = timeout == 0
+
+      begin
+        @parent.wait_for_exists(timeout)
```

Comment:
A previous implementation used the Timeout class to set a total limit on it. The problem is when you set the timeout to 0 (or 1 which wouldn't be handled as a special case) and the combination of calls times it out when it shouldn't.

I decided that this is unlikely to be a real world problem. Yes, if it takes 29 seconds for the parent to exist, then 29 seconds for the element to exist, then another 29 seconds for the element to become displayed, then you are in trouble. 

Under most scenarios if the first call passes even after a substantive time, everything else will quickly pass as well. The worst case practical scenario should be an element taking a long time to exist, then never becoming present, which would time out at in under a minute.

All of that being said... I wonder if it would be an issue if we did:

```ruby
  Watir::Wait.until(timeout) {@parent.wait_for_exists(timeout) && exists?}
```
It should still run through everything at least once before failing if the timeout was set to 0 or 1.